ops/model.py

```diff
@@ -1324,7 +1364,228 @@ def list_files(self, path: str, *, pattern: str = None,
             itself: If path refers to a directory, return information about the
                 directory itself, rather than its contents.
         """
-        return self._pebble.list_files(path, pattern=pattern, itself=itself)
+        return self._pebble.list_files(str(path), pattern=pattern, itself=itself)
+
+    def push_path(self,
+                  source_path: typing.Union[StrOrPath, typing.List[StrOrPath]],
+                  dest_dir: StrOrPath):
+        """Recursively push a local path or files to the remote system.
+
+        Pushing is attempted to completion even if errors occur during the process.  All errors are
+        collected incrementally. After copying has completed, if any errors occurred, a single
+        MultiPushPullError is raised containing details for each error.
+
+        Assuming the following files exist locally:
+
+            * /foo/bar/baz.txt
+            * /foo/foobar.txt
+            * /quux.txt
+
+        You could push the following ways:
+
+            # copy one file
+            container.push_path('/foo/foobar.txt', '/dst')
+            # Destination results: /dst/foobar.txt
```

Comment:
Because these new methods can take a list as well.  So if you want to push several files, you can just `push_path(['/foo/bar.txt', '/foo/baz.txt'], '/somewhere')`.  Also if you want to go straight from disk to disk, you can skip the boiler plate code of pull:

```python
# using old pull
with open('file.name', 'w') as f:
    shutil.copyfileobj(container.pull('/foo/bar.log'), f)
# using pull_path
container.pull_path('/foo/bar.log', 'file.name')
```